lib/pbench/test/functional/server/test_put.py

```diff
@@ -274,8 +264,115 @@ def test_index_all(server_client: PbenchServerClient, login_user):
-    def test_delete_all(server_client: PbenchServerClient, login_user):
+
+class TestList:
+    @pytest.mark.dependency(name="list_none", depends=["upload"], scope="session")
+    def test_list_anonymous(self, server_client: PbenchServerClient):
+        """List all datasets without a login.
+
+        We should see only published datasets. We don't care whether there are
+        pre-existing datasets in this case: we'll simply confirm that every one
+        we see is public. We do care that we don't get an empty list, since we
+        uploaded datasets with access public."""
+        datasets = server_client.get_list(metadata=["dataset.access"])
+        count = 0
+
+        for dataset in datasets:
+            assert dataset.metadata["dataset.access"] == "public"
+            count += 1
+
+        assert count > 1
+
+    @pytest.mark.dependency(name="list_all", depends=["upload"], scope="session")
+    def test_list_filter_or(self, server_client: PbenchServerClient, login_user):
+        """Check a simple OR filter list.
+
+        Authorized as our "tester" user, we can see all the datasets we've
+        uploaded. Try a filtered list choosing datasets run with the "fio"
+        script OR the "linpack" script. Only those datasets should appear: we
+        check the list against a glob of our upload source directory."""
+        fio_names = {Dataset.stem(t) for t in TARBALL_DIR.glob("*fio*.tar.xz")}
+        linpack_names = {Dataset.stem(t) for t in TARBALL_DIR.glob("*linpack*.tar.xz")}
+        datasets = server_client.get_list(
+            metadata=["dataset.metalog.pbench.name"],
+            owner="tester",
+            filter=[
+                "^dataset.metalog.pbench.script:fio",
+                "^dataset.metalog.pbench.script:linpack",
+            ],
+        )
+
+        for dataset in datasets:
+            name = dataset.metadata["dataset.metalog.pbench.name"]
+            if name in fio_names:
+                fio_names.remove(name)
+            elif name in linpack_names:
+                linpack_names.remove(name)
+            else:
+                pytest.fail(
+                    f"Dataset {name} should not have been returned by the filtered query"
+                )
+
+        assert len(fio_names) == 0
+        assert len(linpack_names) == 0
+
+    @pytest.mark.dependency(name="list_all", depends=["upload"], scope="session")
+    def test_list_filter_and(self, server_client: PbenchServerClient, login_user):
+        """Check a simple AND filter list.
+
+        Authorized as our "tester" user, we can see all the datasets we've
+        uploaded. Try a filtered list choosing datasets which are public AND
+        were created in 2018.
+
+        NOTE: the original "owner", "access", "start", "end", and "name"
+        filters are supported and implicitly linked as AND; their matching is
+        also more "friendly" in some cases, e.g., being case-insensitive. In
+        this case we're using "dataset.access"."""
```

Comment:
[Per PEP-8](https://peps.python.org/pep-0008/#documentation-strings), the closing triple-double-quote is supposed to be on a line by itself (which would make parsing the end of that last sentence easier 😁 and it would help the eye find the beginning of the code body).